x-pack/plugins/security_solution/public/management/pages/policy/view/event_filters/hooks.ts

```diff
@@ -35,46 +32,35 @@ export function useGetAllAssignedEventFilters(
 }
 
 export function useSearchAssignedEventFilters(
-  policyId?: string,
-  filter?: string
+  eventFiltersService: EventFiltersHttpService,
+  policyId: string,
+  options: { filter?: string; page?: number; perPage?: number }
 ): QueryObserverResult<FoundExceptionListItemSchema, ServerApiError> {
-  const http = useHttp();
-  const eventFiltersService = new EventFiltersHttpService(http);
+  const { filter, page, perPage } = options;
 
   return useQuery<FoundExceptionListItemSchema, ServerApiError>(
-    ['eventFilters', 'assigned', policyId],
+    ['eventFilters', 'assigned', 'search', policyId, options],
```

Comment:
here, every new search term, new page, new items per page, that will trigger a new query.

Reply:
Exactly, but when the params are all the same the query is cached and this is what I want to avoid here. For example:
- I search by a term and start a pagination.
- At some point I want to clear the term and reset the pagination.
- Something might change at this point in the results so I want a fresh ones.

Reply:
in that case, you can invalidate the cache  using the queryClient rather than handling the cache manually in the useQuery.